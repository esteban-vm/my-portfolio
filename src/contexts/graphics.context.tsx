import type { SceneStage } from '@/types'
import * as React from 'react'
import { useMediaQuery } from 'react-responsive'

interface IGraphicsContext {
  stage: {
    current: SceneStage
    change: (value: SceneStage) => void
  }

  scene: {
    isMobile: boolean
    isRotating: boolean
    rotate: () => void
    stop: () => void
  }
}

const GraphicsContext = React.createContext<IGraphicsContext>(null!)

export function GraphicsContextProvider(props: { children: React.ReactNode }) {
  const isMobileScene = useMediaQuery({ maxWidth: 767 })
  const [isRotatingScene, setIsRotatingScene] = React.useState(true)
  const [currentStage, setCurrentStage] = React.useState<SceneStage>(1)

  const rotateScene = React.useCallback(() => setIsRotatingScene(true), [])
  const stopScene = React.useCallback(() => setIsRotatingScene(false), [])

  const value = React.useMemo((): IGraphicsContext => {
    const stage: IGraphicsContext['stage'] = {
      current: currentStage,
      change: setCurrentStage,
    }

    const scene: IGraphicsContext['scene'] = {
      isMobile: isMobileScene,
      isRotating: isRotatingScene,
      rotate: rotateScene,
      stop: stopScene,
    }

    return { stage, scene }
  }, [currentStage, isRotatingScene, rotateScene, isMobileScene, stopScene])

  return <GraphicsContext.Provider value={value} {...props} />
}

export const useGraphicsContext = () => React.useContext(GraphicsContext)
