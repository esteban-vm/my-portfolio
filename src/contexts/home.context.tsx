import type { SceneStage } from '@/types'
import type { ReactNode } from 'react'
import { createContext, useCallback, useContext, useMemo, useState } from 'react'

interface IHomeContext {
  stage: {
    current: SceneStage
    change: (value: SceneStage) => void
  }

  scene: {
    isRotating: boolean
    rotate: () => void
    stop: () => void
  }
}

const HomeContext = createContext<IHomeContext>(null!)

export function HomeContextProvider(props: { children: ReactNode }) {
  const [currentStage, setCurrentStage] = useState<SceneStage>(1)
  const [isRotatingScene, setIsRotatingScene] = useState(true)

  const rotateScene = useCallback(() => setIsRotatingScene(true), [])
  const stopScene = useCallback(() => setIsRotatingScene(false), [])

  const value = useMemo((): IHomeContext => {
    const stage: IHomeContext['stage'] = {
      current: currentStage,
      change: setCurrentStage,
    }

    const scene: IHomeContext['scene'] = {
      isRotating: isRotatingScene,
      rotate: rotateScene,
      stop: stopScene,
    }

    return { stage, scene }
  }, [currentStage, isRotatingScene, rotateScene, stopScene])

  return <HomeContext.Provider value={value} {...props} />
}

export const useHomeContext = () => useContext(HomeContext)
