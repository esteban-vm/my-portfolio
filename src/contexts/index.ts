'use client'

export * from './scene_context'
