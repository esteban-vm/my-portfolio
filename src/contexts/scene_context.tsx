import type { ReactNode } from 'react'
import { createContext, useContext, useState } from 'react'
import { useMediaQuery } from 'react-responsive'

interface ISceneContext {
  isMobile: boolean
  currentStage: AppTypes.CurrentStage
  setCurrentStage: (currentStage: AppTypes.CurrentStage) => void
  isAnimated: boolean
  setIsAnimated: (isAnimated: boolean) => void
}

const SceneContext = createContext<ISceneContext>(null!)

export function SceneContextProvider(props: { children: ReactNode }) {
  const isMobile = useMediaQuery({ maxWidth: 767 })
  const [currentStage, setCurrentStage] = useState<AppTypes.CurrentStage>(1)
  const [isAnimated, setIsAnimated] = useState(true)

  const value: ISceneContext = {
    isMobile,
    currentStage,
    setCurrentStage,
    isAnimated,
    setIsAnimated,
  }

  return <SceneContext.Provider value={value} {...props} />
}

export const useSceneContext = () => useContext(SceneContext)
