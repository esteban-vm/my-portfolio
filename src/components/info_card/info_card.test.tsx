import { SceneContextProvider } from '@/contexts'
import { render } from '@/tests'
import InfoCardContent from './info_card.content'

describe(`${InfoCardContent.name}:`, () => {
  it('should render correctly without passing href', () => {
    const { asFragment } = render(
      <SceneContextProvider>
        <InfoCardContent mainText='Test' />
      </SceneContextProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly passing href', () => {
    const { asFragment } = render(
      <SceneContextProvider>
        <InfoCardContent href='/' mainText='Test' />
      </SceneContextProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
