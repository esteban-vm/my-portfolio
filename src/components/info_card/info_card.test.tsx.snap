// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InfoCardContent: > should render correctly passing href 1`] = `
<DocumentFragment>
  <div
    class="absolute left-1/2 top-[14vh] z-10 -translate-x-1/2 animate-blink rounded-md bg-neon-green-dark text-center font-semibold uppercase text-black ~w-80/[36rem] ~p-3/5 hover:paused md:top-[15vh] lg:top-[16vh]"
  >
    <div
      class="relative"
    >
      <p
        class="line-clamp-3 text-pretty ~text-xs/base"
      >
        Test
      </p>
      <a
        class="absolute left-1/2 top-full w-[90%] -translate-x-1/2 rounded-md bg-black/60 font-bold text-neon-green-darker shadow-md shadow-neon-green-light backdrop-blur-sm transition-all ~px-2/4 ~py-1/3 hover:opacity-95 active:scale-90"
        href="/"
      >
        <span
          class="align-middle ~text-2xs/sm"
        >
          Learn more…
        </span>
        <svg
          aria-label="Learn more…"
          class="inline fill-neon-green-darker ~size-4/5"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 9h6V5l7 7-7 7v-4H6V9z"
          />
        </svg>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`InfoCardContent: > should render correctly without passing href 1`] = `
<DocumentFragment>
  <div
    class="absolute left-1/2 top-[14vh] z-10 -translate-x-1/2 animate-blink rounded-md bg-neon-green-dark text-center font-semibold uppercase text-black ~w-80/[36rem] ~p-3/5 hover:paused md:top-[15vh] lg:top-[16vh]"
  >
    <div
      class="relative"
    >
      <p
        class="line-clamp-3 text-pretty ~text-xs/base"
      >
        Test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
